common/tools/eslint-plugin-azure-sdk/src/rules/ts-doc-internal-private-member.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @file Rule to check if a private class member is tagged with @internal
+ * @author Hamsa Shankar
+ */
+
+import { Modifier, SyntaxKind } from "typescript";
+import { ParserServices, TSESTree } from "@typescript-eslint/experimental-utils";
+import { Node } from "estree";
+import { ParserWeakMapESTreeToTSNode } from "@typescript-eslint/typescript-estree/dist/parser-options";
+import { Rule } from "eslint";
+import { getRuleMetaData } from "../utils";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/**
+ * Helper method for reporting on a node
+ * @param node the Node being operated on
+ * @param context the ESLint runtime context
+ * @param converter a converter from TSESTree Nodes to TSNodes
+ * @param typeChecker the TypeScript TypeChecker
+ * @throws if the Node passes throught the initial checks and has an internal tag
+ */
+const reportInternal = (
+  node: Node,
+  context: Rule.RuleContext,
+  converter: ParserWeakMapESTreeToTSNode,
+): void => {
+  const tsNode = converter.get(node as TSESTree.Node) as any;
+
+  const modifiers = converter.get(node as TSESTree.Node).modifiers;
+  // if type is internal and has a TSDoc and is a private member
+  if (tsNode.jsDoc !== undefined
+    && modifiers !== undefined && modifiers.some(
+      (modifier: Modifier): boolean => modifier.kind === SyntaxKind.PrivateKeyword)
+  ) {
```

Comment:
```suggestion
    && modifiers?.some((modifier) => modifier.kind === SyntaxKind.PrivateKeyword)) {
```

The types here are trivially inferred from context, so there shouldn't be any need to specify them, and `modifiers?.some` will take care of checking that `modifiers` is defined for you.

Reply:
Awesome thanks